core/src/main/scala/org/apache/spark/ExecutorAllocationManager.scala

```diff
@@ -920,7 +931,7 @@ private[spark] class ExecutorAllocationManager(
       val attempts = resourceProfileIdToStageAttempt.getOrElse(rp, Set.empty).toSeq
       // attempts is a Set, change to Seq so we keep all values
       attempts.map { attempt =>
-        stageAttemptToNumRunningTask.getOrElseUpdate(attempt, 0)
+        stageAttemptToNumRunningTask.getOrElse(attempt, 0)
```

Comment:
Last item - is the point here that there's no point is storing a 0 in the map, as that's the default, and it makes the attempt turn up in the map keys, when you need it not to?

Reply:
I would like to clarify this, is this causing a leak?  ie are you getting some event late or out of order such that this added an entry in stageAttemptToNumRunningTask but then its never removed?    I think this change is actually ok but would like to know if its needed. I would also like to know when it happened to see if perhaps there is a race we can handle elsewhere in the stage attempt being added.

Reply:
guess there is more in the jira:
> If spark-dynamic-executor-allocation thread calls schedule() after a SparkListenerTaskEnd event for the last task in a stage
but before SparkListenerStageCompleted event for the stage, then stageAttemptToNumRunningTask will not be cleaned up properly.

I'm not following this though because onTaskEnd both removes the stage from stageAttemptToNumRunningTask as well as removes the stageAttempt from resourceProfileIdToStageAttempt. so when this is called it shouldn't be in  resourceProfileIdToStageAttempt.  Is there a case we are hitting that it is in there, like multiple stages using that resource profile?

Reply:
ok understanding the other issue, makes this one clear now, this change is fine. 
this introduces the possibility to add it back to the map with value 0 and it never be removed, returning 0 here is fine because it is really needs to be added back it will be added in the task start handling